stetho/src/main/java/com/facebook/stetho/inspector/protocol/ScreencastDispatcher.java

```diff
@@ -0,0 +1,142 @@
+
+public class ScreencastDispatcher {
+  private final Handler mMainHandler = new Handler(Looper.getMainLooper());
+  private final HandlerThread mHandlerThread = new HandlerThread("Screencast Thread");
+  private final BitmapFetchRunnable mBitmapFetchRunnable = new BitmapFetchRunnable();
+  private final ActivityTracker mActivityTracker = ActivityTracker.get();
+  private final EventDispatchRunnable mEventDispatchRunnable = new EventDispatchRunnable();
+  private final ScreencastFrameEvent mEvent = new ScreencastFrameEvent();
+  private final ScreencastFrameEvent.Metadata mMetadata = new ScreencastFrameEvent.Metadata();
+  private boolean mIsRunning;
+  private static final float FRAMERATE = 60f;
+  private static final long FRAME_DELAY = (long) (1000 / FRAMERATE);
+
+  private Handler mBackgroundHandler;
+  private JsonRpcPeer mPeer;
+
+  public ScreencastDispatcher() {
+    mHandlerThread.start();
+  }
+
+  public void startScreencast(JsonRpcPeer peer) {
+    LogUtil.d("Starting screencast");
+    mPeer = peer;
+    mIsRunning = true;
+    mBackgroundHandler = new Handler(mHandlerThread.getLooper());
+    mMainHandler.postDelayed(mBitmapFetchRunnable, FRAME_DELAY);
+  }
+
+  public void stopScreencast() {
+    LogUtil.d("Stopping screencast");
+    mMainHandler.removeCallbacks(mBitmapFetchRunnable);
+    mBackgroundHandler.removeCallbacks(mEventDispatchRunnable);
+    mIsRunning = false;
+  }
+
+  private class BitmapFetchRunnable implements Runnable {
+    @Override
+    public void run() {
+      mBackgroundHandler.post(mEventDispatchRunnable
+          .withBitmap(getScreenBitmap())
+          .withEndAction(this));
+    }
+
+    private Bitmap getScreenBitmap() {
+      if (!mIsRunning) {
+        return null;
+      }
+      Activity activity = mActivityTracker.tryGetTopActivity();
+      if (activity == null) {
+        return null;
+      }
+      // This stuff needs to happen in the UI thread
+      View rootView = activity.getWindow().getDecorView();
+      rootView.setDrawingCacheEnabled(true);
+      Bitmap drawingCache = rootView.getDrawingCache();
+      Bitmap bitmap = android.graphics.Bitmap.createBitmap(drawingCache);
+      rootView.setDrawingCacheEnabled(false);
+      return bitmap;
+    }
+  }
+
+  private class EventDispatchRunnable implements Runnable {
+    private Bitmap mBitmap;
+    private Runnable mEndAction;
+
+    private EventDispatchRunnable withBitmap(Bitmap bitmap) {
+      mBitmap = bitmap;
+      return this;
+    }
+
+    private EventDispatchRunnable withEndAction(Runnable endAction) {
+      mEndAction = endAction;
+      return this;
+    }
+
+    @Override
+    public void run() {
+      if (!mIsRunning) {
+        return;
+      }
+      ByteArrayOutputStream stream = new ByteArrayOutputStream();
+      mBitmap.compress(Bitmap.CompressFormat.JPEG, 80, stream);
+      mEvent.data = Base64.encodeToString(stream.toByteArray(), Base64.DEFAULT);
+      mMetadata.offsetTop = 0;
```

Comment:
Since this is a very hot code path, I decided to reuse the event instance, otherwise we'd be creating 2 new instances per frame. I had it create it every time in the first version, but once I changed it to reuse the instances, I observed a noticeable improvement in the screencasting performance.